javascripts/discourse/initializers/initialize-for-header-icon-links.gjs

```diff
@@ -1,22 +1,16 @@
 import { withPluginApi } from "discourse/lib/plugin-api";
-import { iconNode } from "discourse-common/lib/icon-library";
+import { iconHTML } from "discourse-common/lib/icon-library";
 import { dasherize } from "@ember/string";
 import isValidUrl from "../lib/isValidUrl";
-import { h } from "virtual-dom";
+import { htmlSafe } from "@ember/template";
 
-function buildIcon(icon) {
-  if (isValidUrl(icon)) {
-    return h(
-      "img",
-      {
-        attributes: {
-          src: icon,
-        },
-      },
-      ""
-    );
+function buildIcon(iconNameOrImageUrl) {
+  if (isValidUrl(iconNameOrImageUrl)) {
+    return <template>
+      <img src="{{iconNameOrImageUrl}}" alt="This should be something, shouldn't it?" />
+    </template>
   } else {
-    return iconNode(icon.toLowerCase());
+    return htmlSafe(iconHTML(iconNameOrImageUrl.toLowerCase()));
```

Comment:
Using an htmlSafe'd iconHTML allows us to embed the icon in a Glimmer template, but is this the best way to do it?

Reply:
How about something like `return <template>{{icon iconNameOrImageUrl}}</template>` ?

(`icon` helper imported from `discourse-common/helpers/d-icon`)

Reply:
Nice! Done.